src/Catel.Core/Catel.Core.Shared/Services/ObjectIdGenerator.cs

```diff
@@ -0,0 +1,72 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="ObjectIdGenerator.cs" company="Catel development team">
+//   Copyright (c) 2008 - 2018 Catel development team. All rights reserved.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+
+namespace Catel.Services
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// The ObjectIdGenerator class.
+    /// </summary>
+    /// <typeparam name="TObjectType">The object type</typeparam>
+    /// <typeparam name="TUniqueIdentifier">The unique identifier type</typeparam>
+    public abstract class ObjectIdGenerator<TObjectType, TUniqueIdentifier> : IObjectIdGenerator<TObjectType, TUniqueIdentifier>
+    {
+        private static Queue<TUniqueIdentifier> _releasedUniqueIdentifiers;
+
+        private static readonly object _syncObj = new object();
+
+        /// <inheritdoc />
+        public TUniqueIdentifier GetUniqueIdentifier()
```

Comment:
This one can be removed, it's not a breaking change.

Reply:
Fixed